subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -18,6 +18,20 @@ LOG_MODULE_REGISTER(net_sock_tls, CONFIG_NET_SOCKETS_LOG_LEVEL);
 #include <syscall_handler.h>
 #include <sys/fdtable.h>
 
+/* TODO: Remove all direct access to private fields.
```

Comment:
`mbedtls_2to3.py` from https://github.com/ARMmbed/mbedtls/pull/4800 may help to locate all the places where a private field is needed. If there are fields that you can't avoid accessing, please raise an issue on the mbedtls project (if there isn't one already).